src/lib/getKeyDownHandlers.ts

```diff
@@ -38,7 +38,11 @@ const getKeyDownHandlers = (
             componentActionHandlers[actionName],
             componentPartKeyAction[actionName].keyCombinations,
           )
-          eventHandler && eventHandler(event)
+
+          if (eventHandler) {
+            event.preventDefault()
```

Comment:
having concerns about this line. In FocusZone ```onKeyDown``` handler checks if the event is ```preventDefaulted```, and if yes, it will do nothing. Just to imagine, let's say the parent container of the component is wrapped with FocusZone, and the same key event is propagated to FocusZone, it won't work as expected.

Indeed, in some cases we need to preventDefault.  If you look at ```chatMessageBehavior```, we have done it using keyActions

```js
keyActions: {
    root: {
      // prevents default FocusZone behavior, in this case, prevents using arrow keys as navigation (we only want a Tab key to navigate)
      preventDefault: {
        keyCombinations: [{ keyCode: keyboardKey.ArrowUp }, { keyCode: keyboardKey.ArrowDown }],
      },
    },
  },
```

So to wrap up my thought, it's not really safe to ```preventDefault``` for all the cases.
We consider making it optional, by passing extra param or using key actions as described above.

Please, let me know what you think.

Reply:
this change is introduced to prevent scrolling of the page when one would click up-down keys for selectable List (please, take a look on the corresponding example on the Stardust page we have now to see this problem) - each keypress results not just in switching list item, but also in scrolling the page. This is not a desired effect for sure.

Solutions? While we might consider to make `preventDefault()` calls for each action handler in the List component, it seems to be redundant. The logic where we have check that client has provided handler for the event seems to be a clear suggestion that this event will be handled by client. So, **by default I would definitely expect that event with handler defined will stop bubbling at the point where it is handled**.

> Just to imagine, let's say the parent container of the component is wrapped with FocusZone, and the same key event is propagated to FocusZone, it won't work as expected.

Sorry, could you, please, suggest what the case is - cannot understand what this is about. Currently have the following picture from these words, but not sure that have got an idea. Maybe you could suggest code example, or further elaborate on the case youare talking about?

```jsx
<FocusZone>
  <Container>
     <Element />  <--- here is where event raised
  </Container>
</FocusZone>
```

Reply:
Let me try to explain from a different side.
I've tried to make this change in the code and tested the chat prototype. Chat is no more navigable with arrow keys.

The reason for that:

Chat has defined ```keyActions``` in ```chatBehavior```
```js
 focus: {
        keyCombinations: [{ keyCode: keyboardKey.ArrowLeft }],
      },
```
When we press Arrow up or down on a chat, next happens:
- ```onKeyDown``` handler is triggered in ```getKeyDownHandler.ts```
- after mapping action names to key actions, we've got our handler, which contains filtering logic with keys inside it
- ```event.preventDefault``` is called for any keyboard event and any key
- the event goes to FocusZone where it checks next
```js
if (ev.isDefaultPrevented()) {
      return undefined
    }
``` 
- so navigation won't work in FocusZone

If we want to stick with ```event.preventDefault```, we have to move it to ```keyboardHandlerFilter.ts```, line 19, so it will look like
```js
if (isHandled) {
    event.preventDefault()
    handler(event)
  }
```

So then  ```event.preventDefault``` will be called only for specific keys, when the keyboard handler is triggered.

But still, I am afraid if we miss something and it can break some other use case. I might vote for having it as an optional param.

 @kuzhelov  Let me know what you think and if you get the idea or we can talk about it

Reply:
preventDefault should not be there by default. It should be part of each event handler that requires it. I understand that it requires users to write a bit more code and have this understanding, but it also gives them flexibility in how they want to write their handlers and where do they want to catch events.

Reply:
agree, let me address that